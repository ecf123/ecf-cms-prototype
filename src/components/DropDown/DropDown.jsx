import "./DropDown.scss";
import topicArrow from "../../assets/upside-down-arrow-grey.svg";

const DropDown = ({topicTitle}) => {
  return (
    <div className="dropDown">
        <div className="dropDownTopics">
            <h3>{topicTitle}</h3>
            <img src={topicArrow} alt="" />
            <div className="dropDownLessons"></div>
        </div>
    </div>
  )
}

export default DropDown