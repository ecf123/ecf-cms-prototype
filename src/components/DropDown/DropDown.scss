@use "../../styles/main.scss" as *;

.drop-down {
  max-width: 55.5rem;
  font-family: $font-secondary;
  letter-spacing: $letter-spacing;
  font-size: $font-size-small;
  font-weight: $font-weight-regular;

  &__title {
    display: flex;
    justify-content: space-between;
    padding: 1.9rem;
    cursor: pointer;
    background: $color-white-smoke;
    color: $color-blue-navy-dark;
  }

  &__title::after {
    float: right;
    right: 10%;
  }

  &__list {
    display: flex;
    clear: both;
    width: 100%;
    background-color: $color-white;
    height: 0;
    overflow: hidden;
    transition: height 0.4s ease;
  }

  &__list li {
    padding: 1.9rem;
  }
  &__list li a {
    text-decoration: none;
  }

  &__content {
    color: $color-grey-light;
  }

  &__image {
    margin-right: 1.25rem;
  }
}

#click {
  position: absolute;
  opacity: 0;
  height: 0;
}

#click ~ label span .drop-down__arrow {
  transition: all ease-in-out 0.3s;
}

#click:checked ~ label span .drop-down__arrow {
  transform: rotate(180deg);
}

#click:checked ~ .drop-down__list {
  height: auto;
}
