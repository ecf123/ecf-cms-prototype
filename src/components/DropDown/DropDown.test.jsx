import { customRender } from "../../utils/testUtils";
import { render, screen } from "@testing-library/react";
import { fireEvent, getByText } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import DropDown from "./DropDown";

it("Should render DropDown component", () => {
  const { container } = customRender(<DropDown title={"Business"} lessonTitle="Financial Planning with Barclays" />);

  expect(container).toMatchSnapshot();
});

it("Should check if the checkbox has been checked", () => {
  render(<DropDown title={"Business"} lessonTitle="Financial Planning with Barclays" />);

  userEvent.click(screen.getByRole("checkbox"));

  expect(screen.getByRole("checkbox")).toBeChecked();
});

it("Should display lesson list if checkbox is checked", () => {

  const { container } = customRender(<DropDown title={"Business"} lessonTitle="Financial Planning with Barclays" />);

  let contentDiv = document.getElementsByClassName("drop-down__list");
let style = window.getComputedStyle(contentDiv[0]);
  expect(style.height).toBe("0px");

  fireEvent.click(screen.getByRole("checkbox"));

  contentDiv = document.getElementsByClassName("drop-down__list");
  style = window.getComputedStyle(contentDiv[0]);
  expect(style.height).toBe("auto");

  
});
