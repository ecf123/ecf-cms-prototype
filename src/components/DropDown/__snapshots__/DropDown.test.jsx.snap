// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render DropDown component 1`] = `
<div>
  <nav
    class="drop-down"
  >
    <input
      id="click"
      type="checkbox"
    />
    <label
      for="click"
    >
      <span
        class="drop-down__title"
      >
        <img
          alt=""
          class="drop-down__arrow"
          src="upside-down-arrow-grey.svg"
        />
      </span>
    </label>
    <ul
      class="drop-down__list"
    >
      <li>
        <a
          href="#"
        >
          <p
            class="drop-down__content"
          >
            <img
              alt=""
              class="drop-down__image"
            />
            Financial Planning with Barclays
          </p>
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
